mne/channels/channels.py

```diff
@@ -740,7 +740,23 @@ def pick_types(self, meg=None, eeg=False, stim=False, eog=False,
             ias=ias, syst=syst, seeg=seeg, dipole=dipole, gof=gof, bio=bio,
             ecog=ecog, fnirs=fnirs, include=include, exclude=exclude,
             selection=selection)
-        return self._pick_drop_channels(idx)
+
+        res = self._pick_drop_channels(idx)
+
+        # remove dropped channel types from reject and flat
+        if getattr(self, 'reject', None) is not None:
+            ch_types = list(self.reject.keys())
+            for ch_type in ch_types:
+                if ch_type not in self:
+                    del self.reject[ch_type]
+
+        if getattr(self, 'flat', None) is not None:
+            ch_types = list(self.flat.keys())
+            for ch_type in ch_types:
+                if ch_type not in self:
+                    del self.flat[ch_type]
+
+        return res
```

Comment:
Should be able to simply use
```suggestion
        return self
```